core/src/main/java/org/lflang/analyses/statespace/EventQueue.java

```diff
@@ -0,0 +1,22 @@
+/**
+ * An event queue implementation that sorts events by time tag order
+ *
+ * @author{Shaokai Lin <shaokai@berkeley.edu>}
+ */
+package org.lflang.analyses.statespace;
+
+import java.util.PriorityQueue;
+
+public class EventQueue extends PriorityQueue<Event> {
```

Comment:
I would suggest not extending this class but hiding it inside. As it is you inherit also the `offer` method which inserts an element, but does not constrain uniqueness as your implementation claims.